### PR TITLE
feat: Enable support to specify database version

### PR DESCRIPTION
**Pull request by rajc242**
Provide support to specify database version. Currently default value has been set to "POSTGRES_14". Please check out our [documentation](https://registry.terraform.io/providers/hashicorp/google/5.16.0/docs/resources/alloydb_cluster) for more details.

### CODE REVIEW

**Comment by imrannayer** [comment]
@rajc242 thanks for the PR. Can you plz update version.tf file with >5.13?

File: examples/simple_example/main.tf [hunk 1]
```diff
@@ -69,7 +69,9 @@ module "alloy-db" {
     instance_id = "primary-instance-1",
     client_connection_config = {
       require_connectors = false
-      ssl_config         = "ALLOW_UNENCRYPTED_AND_ENCRYPTED"
+      ssl_config = {
+        ssl_mode = "ALLOW_UNENCRYPTED_AND_ENCRYPTED"
+      }
```
Comment (maksym-iv) [inline examples/simple_example/main.tf:74]: Shouldn't it be just 
```hcl
ssl_mode = `"ALLOW_UNENCRYPTED_AND_ENCRYPTED"`
```
?

Source seems not using the `ssl_config` attribute
```hcl
  dynamic "client_connection_config" {
    for_each = lookup(var.primary_instance, "ssl_mode", null) != null || lookup(var.primary_instance, "require_connectors", null) != null ? ["client_connection_config"] : []

    content {
      require_connectors = try(var.primary_instance.require_connectors, null)
      ssl_config {
        ssl_mode = try(var.primary_instance.ssl_mode, null)
      }
    }
  }
```

Comment (maksym-iv) [reply examples/simple_example/main.tf:74]: But in the actual module `ssl_config` is not used 
* https://github.com/GoogleCloudPlatform/terraform-google-alloy-db/blob/68e3aeac524d9eebeba17e58c0e451d6b9351dc0/main.tf#L158-L167
* https://github.com/GoogleCloudPlatform/terraform-google-alloy-db/blob/68e3aeac524d9eebeba17e58c0e451d6b9351dc0/main.tf#L195-L203

Comment (imrannayer) [reply examples/simple_example/main.tf:74]: @rajc242 it should be 
```
ssl_mode = `"ALLOW_UNENCRYPTED_AND_ENCRYPTED"`
```

Comment (rajc242) [reply examples/simple_example/main.tf]: Fix applied.

**Review by imrannayer — APPROVED** [review]

**Pull request merged**